agent/handlers/utils/helpers.go

```diff
@@ -129,3 +131,10 @@ func LimitReachedHandler(auditLogger audit.AuditLogger) func(http.ResponseWriter
 		auditLogger.Log(logRequest, http.StatusTooManyRequests, "")
 	}
 }
+
+// WriteErrorToDefaultLogger write the requestType, httpStatusCode and responseJSON of error to the default logger
+func WriteErrorToDefaultLogger(httpStatusCode int, responseJSON []byte, requestType string) {
+	if httpStatusCode >= 400 && httpStatusCode <= 599 {
```

Comment:
would it be better to have `if httpStatusCode<200 || httpStatusCode>299` ?

Reply:
As httpStatusCode 1xx and 3xx would not be considered as an error code; therefore the if statement `httpStatusCode >= 400 && httpStatusCode <= 599` is used here to only propagate httpStatusCode 4xx and 5xx to the agent log.